### ISSUE TITLE
Running make in pwsafe/install/deb/Makefile => "No rule to make target.."

### ISSUE THREAD

**Issue by libBletchley** [pwsafe/pwsafe#347]
Doing this:
$ `cd pwsafe/install/deb`
$ `make all`
Results in:
```
make: *** No rule to make target '../../src/ui/wxWidgets/GCCUnicodeRelease/pwsafe', needed by 'passwordsafe-debian-0.94.amd64.deb'.  Stop.
```
The reason I went down that path is because these installation instructions are wrong:
```
3. Installation on Debian or Ubuntu:
3.1 Download the .deb file that corresponds to your distribution.
3.2 Install it using dpkg:
$ sudo dpkg -i passwordsafe-*.deb
Dpkg will complain if prerequisite packages are missing. To install
the missing packages:
$ sudo aptitude install missing-pkgX missing-pkgY
```
Step 3.1 fails b/c there are no .deb files.  Or perhaps I'm just looking in the wrong places, in which case the documentation should give a clue about where to find them.

These instructions suggest attempting an install and then tending to errors.  The dpkg tool can be told to show missing dependencies (and perhaps install them automatically - not sure), before installing the target package.  That would be more elegant to document.

(edit)
I found the `release` page (which btw was hard to find because it isn't linked to from https://github.com/pwsafe/pwsafe).  So in https://github.com/pwsafe/pwsafe/releases there are `.deb` files that are quite old.  It's no longer available in current releases, so the linux install documentation "README.LINUX" needs to be updated.

**Comment by ronys** [comment]
Thanks for reporting this.

1. For manual compilation, you should run make from the toplevel, and not from the install/deb directory.
2. Can you update the README.LINUX file and send me a pull request, or just mail me the updated version for inclusion in the next release (coming soon)?

Cheers,

  Rony

**Comment by libBletchley** [comment]
> For manual compilation, you should run make from the toplevel, and not from the install/deb directory.

There is no `GCCUnicodeRelease` subdir in the whole tree, so when I run `make -f install/deb/Makefile all` from the top level, the same error occurs.

> Can you update the README.LINUX file and send me a pull request, or just mail me the updated version for inclusion in the next release (coming soon)?

Sure, once I work out what goes in there.  ATM it looks like the debian route is hosed.

**Comment by libBletchley** [comment]
I also tried simply running `make` at the top level, neglecting the debian-specialized Makefile.  I got some errors which lead me to run `aptitude install gettext wx-common'.  Then I still get an error:

```
g++  -fPIC -Wall -I../../core -I../.. `/usr/bin/wx-config --debug=no --unicode=yes --inplace --cxxflags`   -fPIC -Wall -I../../core -I../.. `/usr/bin/wx-config --debug=no --unicode=yes --inplace --cxxflags`  -std=c++11 -O0 -g -ggdb -DUNICODE -DDEBUG -O0 -g -ggdb -DUNICODE -DDEBUG -c run.cpp -o ../../../obj/unicodedebug/os/run.o
/bin/sh: 1: /usr/bin/wx-config: not found
/bin/sh: 1: /usr/bin/wx-config: not found
run.cpp:14:24: fatal error: wx/process.h: No such file or directory
 #include <wx/process.h>
                        ^
compilation terminated.
Makefile:79: recipe for target '../../../obj/unicodedebug/os/run.o' failed
make[2]: *** [../../../obj/unicodedebug/os/run.o] Error 1
make[2]: Leaving directory '/opt/src/pwsafe/src/os/unix'
Makefile:84: recipe for target 'unicodedebug' failed
make[1]: *** [unicodedebug] Error 2
make[1]: Leaving directory '/opt/src/pwsafe/src/os/unix'
Makefile.linux:50: recipe for target 'unicodedebug' failed
make: *** [unicodedebug] Error 2
```

I was somewhat expecting the `wx-common` package to include this `/usr/bin/wx-config` utility, but apparently not.

**Comment by ronys** [comment]
Please read https://github.com/pwsafe/pwsafe/blob/master/README.LINUX.DEVELOPERS.txt

**Comment by libBletchley** [comment]
Thanks ronys.

On that page, there is a blurb about the need to install wxwidgets, generically saying to install "wxWidgets 3.0".  But there is no such package.  wxWidgets is actually a large collection of packages:
```
                 libwxbase3.0-0-unofficial
                 libwxbase3.0-dev
                 libwxgtk3.0-0-unofficial
                 libwxgtk3.0-dev
                 wx3.0-headers
                 wx-common
                 libwxbase3.0-dbg
                 libwxgtk3.0-dbg
                 wx3.0-i18n
                 wx3.0-examples
                 wx3.0-doc
```
The http://codelite.org/LiteEditor/WxWidgets30Binaries#toc2 page says "the first 6 of those are required; the others optional (and of decreasing importance). "  How can that document make a generic statement like that, without knowing that users are building pwsafe?  Which of these packages are needed for pwsafe?

**Comment by ronys** [comment]
I can't speak for codelite, but the README.LINUX.DEVELOPERS.txt is pretty specific.

You might find it easier to run the relevant  Misc/setup-*-dev-env.sh script described in the document.

**Comment by libBletchley** [comment]
I just noticed your reply, after I overcame the issue you're replying to.  Sorry for the confusion.

So when I was following README.LINUX.DEVELOPERS.txt as a debian user, I reached the Fedora section and thought that was it.  Then guessed that I need to do a `make deb`.  It turns out those instructions go back to directing debian users further down the document.  So far those instructions seem to be working.
